crates/rust-lib/src/lib.rs

```diff
@@ -470,6 +472,11 @@ pub trait RustGenerator<'a> {
         } else {
             mode
         };
+        // Lists with `own` handles must always be owned
+        let mode = match *ty {
+            Type::Id(id) if self.info(id).has_own_handle => TypeMode::Owned,
+            _ => mode,
+        };
```

Comment:
I'm curious why lists are special, necessitating special code in `print_list`. Wouldn't the same issue come up in records and variants?

Reply:
I know I sounds like a broken record but this was basically the mole that came up that I figured I'd whack.

The Rust generator tries to keep things borrowed where it can but that support has not aged well. This, AFAIK, is only here now to handle the non-"owning" mode, which is a non-default setting, where types may be duplicated to aggressively borrow things. Bare types already have special logic for handles and this is sort of copying said special logic to lists-of-handles as well since "this requries owned" is a transitive property through the list.

Not a great solution by any means for sure, and I'm hoping that we can basically just build a corpus of programs to keep working into the future.

Reply:
Makes sense, thanks.